airflow/models/baseoperator.py

```diff
@@ -1001,6 +1001,17 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         SetupTeardownContext.set_work_task_roots_and_leaves()
 
+    @staticmethod
+    def add_task_to_context(task):
```

Comment:
what what if `__enter__` returns a setup / teardown context object and not the task itself.

then the naming could be better:
```
with s >> t as ctx:
    ctx.add_task(w)
```
wdyt?

Reply:
Nice! Done

Reply:
nice.  now here's a thought.... I think SetupTeardownContext should be private class, not part of the user API.  Or at least we should severely constrain which parts of it are public.
So it occurs to me we should either (1) add a new class that is very lightweight and basically only has method `add_task` and return that instead of SetupTeardownContext.... OR... (2) we can maybe rename  SetupTeardownContext to AbstractSetupTeardownContext and make it `:meta private:` and then subclass `SetupTeardownContext` is public and only adds method `add_task` -- so all the methods would be "there" but they would be undocumented and therefore reserved for us -- there's an analogy with abstractoperator / baseoperator here.

Reply:
Done. Can you take a look

Reply:
OK looks good.  

Thinking again... and I'm trying to channel my inner @uranusjr here.... I think that maybe we should call it (the private one) BaseSetupTeardownContext instead of AbstractSetupTeardownContext because, I think Abstract strongly implies that it is not meant to be used directly, but only via subclasses.  Whereas, the word Base seems to better tolerate direct usage -- and we use it directly. That sound right to you?  I know historically it's common to use Base for abstract classes to, but it just seems like the better of the two for something that is used as a concrete class.  Maybe you have a better idea...

Reply:
Sounds good to me too. Updated

Reply:
Super.  Thanks